src/sentry/snuba/models.py

```diff
@@ -97,3 +99,26 @@ class Status(Enum):
     class Meta:
         app_label = "sentry"
         db_table = "sentry_querysubscription"
+
+    # We want the `QuerySubscription` to get properly created in Snuba, so we'll run it through the
+    # purpose-built logic for that operation rather than copying the data verbatim. This will result
+    # in an identical duplicate of the `QuerySubscription` model with a unique `subscription_id`.
+    def write_relocation_import(
+        self, pk_map: PrimaryKeyMap, scope: ImportScope, flags: ImportFlags
+    ) -> Optional[Tuple[int, int, ImportKind]]:
+        # TODO(getsentry/team-ospo#190): Prevents a circular import; could probably split up the
+        # source module in such a way that this is no longer an issue.
+        from sentry.snuba.subscriptions import create_snuba_subscription
+
+        old_pk = super()._normalize_before_relocation_import(pk_map, scope, flags)
+        if old_pk is None:
+            return None
+
+        subscription = create_snuba_subscription(self.project, self.type, self.snuba_query)
+
+        # Keep the original dates.
+        subscription.date_added = self.date_added
+        subscription.date_updated = self.date_updated
```

Comment:
Does it make sense to have a new `date_updated` here instead of using the old value?

Reply:
You're right - I've removed this line.